pkg/dispatcher/prometheus.go

```diff
@@ -30,8 +30,9 @@ type PrometheusAPI interface {
 }
 
 // GetJobVolumesFromPrometheus gets job volumes from a Prometheus server
-func GetJobVolumesFromPrometheus(ctx context.Context, prometheusAPI PrometheusAPI) (map[string]float64, error) {
-	result, warnings, err := prometheusAPI.Query(ctx, `sum(increase(prowjob_state_transitions{state="pending"}[7d])) by (job_name)`, time.Now())
+func GetJobVolumesFromPrometheus(ctx context.Context, prometheusAPI PrometheusAPI, daysBefore int) (map[string]float64, error) {
+	y, m, d := time.Now().Add(-time.Duration(24*(daysBefore+1)) * time.Hour).Date()
```

Comment:
The `+1` makes the interface of this function confusing, and thinking about it, "days before yesterday" is confusing too. How about make this a 1..15 interval (disallow 0), with `1` being "last day" and `15` "last 15 days", then both the semantics and code can be simpler.

Reply:
Move the calculation of time out of the function.
It is always the last 7 days (part of the query).
The time cal is for given a time as now for the last 7 days.